core/src/test/scala/integration/kafka/api/SaslClientsWithInvalidCredentialsTest.scala

```diff
@@ -68,12 +74,27 @@ class SaslClientsWithInvalidCredentialsTest extends IntegrationTestHarness with
   @Test
   def testProducerWithAuthenticationFailure() {
     verifyAuthenticationException(() => sendOneRecord(10000))
+    verifyAuthenticationException(() => producers.head.partitionsFor(topic))
```

Comment:
Nit: we can change the parameter to be `by name` (`=> Unit` instead of  `() => Unit`) and then you don't need the noisy `() =>` in every invocation.